libfs/fs.go

```diff
@@ -366,15 +387,22 @@ func (fs *FS) Join(elem ...string) string {
 
 // TempFile implements the billy.Filesystem interface for FS.
 func (fs *FS) TempFile(dir, prefix string) (billy.File, error) {
-	// We'd have to turn off journaling to support TempFile fully.
-	return nil, errors.New("TempFile is not currently supported")
+	// We'd have to turn off journaling to support TempFile perfectly,
+	// but the given uniq ID and a random number should be good
+	// enough.  Especially since most users will end up renaming the
+	// temp file before journal flushing even happens.
+	r := rand.Int63()
```

Comment:
Oops, good point about seeding.  I can change to crypto/rand -- I just find it annoying to have to initialize a byte buffer before calling it.  But I guess that's easy enough.